lib/engines/content_object_store/app/controllers/content_object_store/content_block/workflow_controller.rb

```diff
@@ -20,15 +20,12 @@ def update
     ContentObjectStore::SchedulePublishingWorker.dequeue(content_block_edition)
 
     new_edition_params = edition_params.merge!(scheduled_publication_params)
-
+    change_dispatcher = params[:schedule_publishing] == "schedule" ? ContentObjectStore::ChangeDispatcher::Schedule.new : ContentObjectStore::ChangeDispatcher::Now.new
```

Comment:
I wonder if you can break this logic down further so you have a `ContentObjectStore::ChangeDispatcher` class that returns `ContentObjectStore::ChangeDispatcher::Schedule` or `ContentObjectStore::ChangeDispatcher::Now` depending on the value of `schedule_publishing`?

Reply:
We could do! I was starting to wonder if I was going overboard with too many abstractions and indirections. Leaving the controller with this responsibility (rather than hiding it away) felt clearer to the reader. I suspect that if there was every a third type of dispatcher then another refactor might be worth the move?

Reply:
Yeah, that's fair 👍